src/decoders/belief_propogation.jl

```diff
@@ -0,0 +1,152 @@
+using SparseArrays
+
+include("abstract_decoder.jl")
+
+struct BeliefPropagationSetup
+  "Log probabilities"
+  log_probabs::Vector{Float64}
+  
+  "Channel probabilities"
+  channel_probs::Vector{Float64}
+  
+  "Bit to check message matrix"
+  bit_2_check::Matrix{Float64}
```

Comment:
why do we need these if we have `sparse_H` and `sparse_HT` in `BeliefPropagationDecoder`

Reply:
`bit_2_check` and `check_2_bit` are temporary for calculating the messages passed. `sparse_H` and `sparse_HT` are for holding the sparse versions of the parity check matrix matrix, to store which edges to pass messages through